File: Source/Maui/Domain/Accounts/Models/Account.cs
namespace Pot.Maui.Domain.Accounts.Models
{
    public sealed record Account
    {
        public long Id { get; set; }
        public string Bsb { get; set; } = string.Empty;
        public string Number { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public double Balance { get; set; }
        public double Reserved { get; set; }
        public double Allocated { get; set; }
        public double DailyAccrual { get; set; }
        public double Available => Balance - Reserved - Allocated;
    }
}

File: Source/Client/pot-react/src/data/accounts/account.ts
import { z } from 'zod';

export const AccountSchema = z.object({
  rowId: z.string(),
  eTag: z.bigint(),
  bsb: z.string(),
  number: z.string(),
  description: z.string(),
  balance: z.number(),
  reserved: z.number(),
  allocated: z.number(),
  dailyAccrual: z.number(),
  available: z.number(),
});

export type Account = z.infer<typeof AccountSchema>;
